showCrime/dailyIncid/management/commands/parse_UCR.py

```diff
@@ -0,0 +1,397 @@
+# coding=utf8
+''' parse_UCR_pdfquery
+
+parse OPD weekly UCR reports
+
+leans heavily on pdfplumber, esp. its TABLE extraction
+	https://github.com/jsvine/pdfplumber says:
+	Works best on machine-generated, rather than scanned, PDFs. Built on pdfminer and pdfminer.six.
+
+Created on Nov 3, 2017
+	update 12 Feb 19
+
+@author: rik
+'''
+
+from collections import defaultdict
+import pickle # cPickle for python2
+from datetime import datetime
+import glob 
+import json
+import os
+import re
+import sys
+
+import pdfplumber
+
+OPD_UCR_DateFormat = '%Y-%b-%d'
+
+IgnoredLabels = ['Part 1 Crimes','THIS REPORT IS HIERARCHY BASED.',
+				'(homicide, aggravated assault, rape, robbery)' ]
+
+LabelUCROrder = [u'Violent Crime Index', 
+				u'Homicide – 187(a)PC', 
+				u'Homicide – All Other *', 
+				u'Aggravated Assault', 
+				# 190212: new labeles
+				# u'Shooting with injury – 245(a)(2)PC', 
+				u'Assault with a firearm – 245(a)(2)PC',
+				# u'Subtotal - Homicides + Injury Shootings',		 
+				u'Subtotal - Homicides + Firearm Assault',
+				
+				u'Shooting occupied home or vehicle – 246PC', 
+				u'Shooting unoccupied home or vehicle – 247PC', 
+				u'Non-firearm aggravated assaults', 
+				u'Rape', 
+				u'Robbery', 
+				u'Firearm', 
+				u'Knife', 
+				u'Strong-arm', 
+				u'Other dangerous weapon', 
+				# u'Residential  robbery – 212.5(A)PC', 
+				u'Residential  robbery – 212.5(a)PC', 
+				# u'Carjacking – 215(A) PC', 
+				u'Carjacking – 215(a) PC', 
+				u'Burglary', 
+				u'Auto', 
+				u'Residential', 
+				u'Commercial', 
+				u'Other (Includes boats, aircraft, and so on)', 
+				u'Unknown', 
+				u'Motor Vehicle Theft', 
+				u'Larceny', 
+				u'Arson', 
+				u'Total' ]
+
+IgnoreStatLbl = ['Author', 'CreateDate', 'ModDate', 'fname', 'fromDate', 'rptDate', 'toDate']
+
+FixFilePat = {re.compile(r'Area(\d)WeeklyCrimeReport11Jun17Jun18.pdf'): '180619_Area %d Weekly Crime Report 11Jun - 17Jun18.pdf'}
+
+def dateConvert(o):
+	if isinstance(o, datetime):
+		return o.strftime('%y%m%d')
+
+def parse_UCR_pdf(inf,rptDate,fdate,tdate,verbose=False):
```

Comment:
It would be easier for me to understand these variables if you spelled them out instead of using abbreviations. e.g. `pdf_filepath`, `receipt_date`, `file_date`, `transaction_date`.